src/Middleware/HttpLogging/src/HttpLoggingMiddleware.cs

```diff
@@ -189,9 +204,20 @@ private async Task InvokeInternal(HttpContext context)
             {
                 context.Request.Body = originalBody;
             }
+
+            if (loggableUpgradeFeature != null)
+            {
+                context.Features.Set(originalUpgradeFeature);
+            }
         }
     }
 
+    private static bool ResponseHeadersNotWrittenYet(ResponseBufferingStream? responseBufferingStream, UpgradeFeatureLoggingDecorator? upgradeFeatureLogging)
+    {
+        return (responseBufferingStream == null || responseBufferingStream.FirstWrite == false) &&
+               (upgradeFeatureLogging == null || !upgradeFeatureLogging.IsUpgradableRequest);
```

Comment:
You also need to check if Upgrade was actually called. Not all WebSocket requests will be upgraded (e.g. it could 401 for auth).